compute_tools/src/compute.rs

```diff
@@ -695,4 +735,115 @@ LIMIT 100",
+                    if already_downloaded_extensions.contains(&ext_name) {
+                        info!(
+                            "extension {:?} already exists, skipping download",
+                            &ext_name
+                        );
+                        return Ok(());
+                    } else {
+                        already_downloaded_extensions.insert(ext_name.clone());
+                    }
+                }
+                extension_server::download_extension(
+                    &ext_name,
+                    &self
+                        .ext_remote_paths
+                        .get()
+                        .expect("error accessing ext_remote_paths")[&ext_name],
+                    remote_storage,
+                    &self.pgbin,
+                )
+                .await
+            }
+        }
+    }
+
+    #[tokio::main]
+    pub async fn prepare_preload_libraries(&self, compute_state: &ComputeState) -> Result<()> {
+        if self.ext_remote_storage.is_none() {
+            return Ok(());
+        }
+        let pspec = compute_state.pspec.as_ref().expect("spec must be set");
+        let spec = &pspec.spec;
+
+        info!("parse shared_preload_libraries from spec.cluster.settings");
+        let mut libs_vec = Vec::new();
+        if let Some(libs) = spec.cluster.settings.find("shared_preload_libraries") {
+            libs_vec = libs
+                .split(&[',', '\'', ' '])
+                .filter(|s| *s != "neon" && !s.is_empty())
+                .map(str::to_string)
+                .collect();
+        }
+        info!("parse shared_preload_libraries from provided postgresql.conf");
+        // that is used in neon_local and python tests
+        if let Some(conf) = &spec.cluster.postgresql_conf {
+            let conf_lines = conf.split('\n').collect::<Vec<&str>>();
+            let mut shared_preload_libraries_line = "";
+            for line in conf_lines {
+                if line.starts_with("shared_preload_libraries") {
+                    shared_preload_libraries_line = line;
+                }
+            }
+            let mut preload_libs_vec = Vec::new();
+            if let Some(libs) = shared_preload_libraries_line.split("='").nth(1) {
+                preload_libs_vec = libs
+                    .split(&[',', '\'', ' '])
+                    .filter(|s| *s != "neon" && !s.is_empty())
+                    .map(str::to_string)
+                    .collect();
+            }
+            libs_vec.extend(preload_libs_vec);
+        }
+
+        info!("Download ext_index.json, find the extension paths");
+        self.prepare_external_extensions(compute_state).await?;
+
+        info!("Downloading to shared preload libraries: {:?}", &libs_vec);
+        let mut download_tasks = Vec::new();
+        for library in &libs_vec {
+            download_tasks.push(self.download_extension(library));
```

Comment:
This code assumes that library name, listed in `shared_preload_libraries` IS the name of extension.
I am not sure if it is always the case.

If this was intentional, please write a comment about this assumption.

cc: @hlinnaka

Reply:
maybe we can have a place in extension index that specifies the archive needed to download for specific libraries (if it is not just their name)